docs/docs/when-to-use-use.md

```diff
@@ -0,0 +1,240 @@
+    
+Any non-standand JavaScript pattern needs an abstraction to use. With components, we use `<AngleBracket />` syntax within a rendering system (which then, in turn, gives us a bunch of other convinient things). With resources in JavaScript `use` is an abstraction over the implementation details that ultimately don't matter to users. 
+
+Imagine why you need [`@service`](https://api.emberjs.com/ember/5.4/functions/@ember%2Fservice/service) for [dependency injection](https://nullvoxpopuli.com/2020-08-08-how-does-di-work) -- it's the glue to hide the details of what would otherwise by very verbose to use without the abstraction.
+
+</details>
+
+Note that all of these examples use a `Component` class, but they could just as well be any native vanilla JavaScript class. See [link](https://ember-resources.pages.dev/funcs/link.link) for more on that.
+
+## `import { use } from 'ember-resources';`
+
+This type of use is for connecting a resource to a JavaScript class.
+
+It has a couple different overloads so that the community can provide feedback on the ergonomics of each over time.
+
+
+<details><summary>Under the hood</summary>
+
+`use` uses [`invokeHelper`](https://api.emberjs.com/ember/release/functions/@ember%2Fhelper/invokeHelper) which means that resources _are_ helpers. It also means that other non-resource helpers could be passed to `use` (note tho that non-resource usage is not supported by this library).
+
+There is a potential for a whole "usable system" here, piggy-backing off of the [helper-manager](https://rfcs.emberjs.com/id/0625-helper-managers/) pattern.
+
+</details>
+
+### `@use`
+
+
+#### "@use thisProperty = DefinedAsThis"
+
+This `@use` decorator _replaces_ the property with a getter that gets the underlying value of the resource. You'll see what happens in non-decorator form later.
+
+<details><summary>looking under the hood</summary>
+
+You could think of what the decorator does in this way:
+
+```js
+class Demo extends Component {
+  @use data = Clock;
+}
+```
+is (at runtime) transformed to:
+```js
+class Demo extends Component {
+  #data1 = invokeHelper(this, Clock);
+  
+  get data() {
+    return getValue(this.#data1);
+  }
+}
+```
+
+Note that the _actual_ return value of `resource` is an internal and private object, and interaction with that object is not supported in user-space.
+
+</details>
+
+In this example, the resource is the value of `data`:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource(() => {
+    /* ... */
+    return 2;
+});
+
+class Demo extends Component {
+    @use data = Clock;
+
+    <template>
+        {{this.data}} prints 2
+    </template>
+}
+```
+
+In this example, the `resource` is still the value returned to `data`:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+function LocalizedClock(locale) {
```

Comment:
`LocalizedClock` takes the argument of `locale`, but then never use it. I know it's not the goal of the exercise, but seems like disconnect. What about:

```
return locale.t(2);
```

Reply:
that would apply more implicit information about the example that's not present, like, what defines the function `t`?

later on do I have `/* format defined here */` and then `format(time)`, maybe that type of thing should be in all these examples

Reply:
Yup, do agree that having to explain extra stuff feels off. Two other possible takes on this:

```js
return new Intl.DateTimeFormat(locale).format(2); // This uses native es6 API
```

or

```js
function LocalizedClock(localisationFunction) {
  return localisationFunction(2);
}
```

Can't find what does `format(time)` refer to, but if the example then is "complete", that'd be IMO better.